lib/build/initializer_plugin.dart

```diff
@@ -273,15 +274,24 @@ class DefaultInitializerPlugin implements InitializerPlugin {
           'And got the following errors: ${result.errors}.');
       return null;
     }
-    var value = result.value.value;
+
+    var value = result.value.toStringValue();
+    if (value != null) value = _stringValue(value);
+
+    if (value == null) value = result.value.toBoolValue();
+    if (value == null) value = result.value.toIntValue();
+    if (value == null) value = result.value.toDoubleValue();
+    if (value == null) value = result.value.toListValue();
+    if (value == null) value = result.value.toMapValue();
+    if (value == null) value = result.value.toSymbolValue();
+    if (value == null) value = result.value.toTypeValue();
```

Comment:
Several of these methods won't actually do what you probably expect:
- toListValue will return a list of DartObject's, not the translated versions of them,
- toMapValue will similarly return a map from DartObject to DartObject,
- toSymbolValue will return a string containing the same code points as the symbol would contain, and
- toTypeValue will return a DartType, not a Type.

I ended up doing something similar for source_gen (https://github.com/dart-lang/source_gen/pull/73). Maybe it should be a utility in analyzer.